airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java

```diff
@@ -58,11 +71,28 @@ public OAuthConsentRead getSourceOAuthConsent(final SourceOauthConsentRequest so
     final Map<String, Object> metadata = generateSourceMetadata(sourceDefinitionIdRequestBody.getSourceDefinitionId());
     final OAuthConsentRead result;
     if (OAuthConfigSupplier.hasOAuthConfigSpecification(spec)) {
+      JsonNode oAuthInputConfigurationForConsent;
+
+      if (sourceDefinitionIdRequestBody.getSourceId() == null) {
+        oAuthInputConfigurationForConsent = sourceDefinitionIdRequestBody.getoAuthInputConfiguration();
+      } else {
+        final SourceConnection hydratedSourceConnection =
```

Comment:
This else clause looks mostly duplicated for source and destination. Is it worth extracting a shared function that takes a `ConnectionSpec` and returns the `JsonNode oAuthInputConfigurationForConsent`?

Reply:
I need to rework this in order to use the advanced oAuth config. Will move it back to draft

Reply:
Done